components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java

```diff
@@ -193,6 +197,9 @@ public void cancelled() {
             } catch (SecretManagementException e) {
                 LOG.error("Error while resolving API key. ", e);
                 asyncReturn.accept(authenticationContext, Collections.emptyMap(), OUTCOME_FAIL);
+            } catch (Exception e) {
+                LOG.error("Error while invoking callChoreo. ", e);
+                asyncReturn.accept(authenticationContext, Collections.emptyMap(), OUTCOME_FAIL);
```

Comment:
Is it okay to pass an immutable map to `accept` function ?

Reply:
We have similarly passed immutable maps through out the code. I adopted the same code.